nixos/modules/i18n/inputMethod/default.xml

```diff
@@ -0,0 +1,99 @@
+<chapter xmlns="http://docbook.org/ns/docbook"
+         xmlns:xlink="http://www.w3.org/1999/xlink"
+         xmlns:xi="http://www.w3.org/2001/XInclude"
+         version="5.0"
+         xml:id="module-services-input-methods">
+
+<title>Input Methods</title>
+
+<para>Input methods are an operating system component that allows any data, such as keyboard strokes or mouse movements, to be received as input. In this way users can enter characters and symbols not found on their input devices. Using an input method is obligatory for any language that has more graphemes than there are keys on the keyboard.</para>
+
+<para>The following input methods are available in NixOS:</para>
+
+<itemizedlist>
+  <listitem><para>IBus: The intelligent input bus.</para></listitem>
+  <listitem><para>Fcitx: A customizable lightweight input method.</para></listitem>
+  <listitem><para>Nabi: A Korean input method based on XIM.</para></listitem>
+  <listitem><para>Uim: The universal input method, is a library with a XIM bridge.</para></listitem>
+</itemizedlist>
+
+<section><title>IBus</title>
+
+<para>IBus is an Intelligent Input Bus. It provides full featured and user friendly input method user interface.</para>
+
+<para>The following snippet can be used to configure IBus:</para>
+
+<programlisting>
+i18n.inputMethod = {
+  enabled = "ibus";
+  ibus.engines = with pkgs.ibus-engines; [ anthy hangul mozc ];
+};
+</programlisting>
+
+<para><literal>i18n.inputMethod.ibus.engines</literal> is optional and can be used to add extra IBus engines.</para>
+
+<para>Available extra IBus engines are:</para>
+
+<itemizedlist>
+  <listitem><para>Anthy (<literal>ibus-engines.anthy</literal>): Anthy is a system for Japanese input method. It converts Hiragana text to Kana Kanji mixed text.</para></listitem>
+  <listitem><para>Hangul (<literal>ibus-engines.hangul</literal>): Korean input method.</para></listitem>
+  <listitem><para>m17n (<literal>ibus-engines.m17n</literal>): m17n is an input method that uses input methods and corresponding icons in the m17n database.</para></listitem>
+  <listitem><para>mozc (<literal>ibus-engines.mozc</literal>): A Japanese input method from Google.</para></listitem>
+  <listitem><para>Table (<literal>ibus-engines.table</literal>): An input method that load tables of input methods.</para></listitem>
+  <listitem><para>table-others (<literal>ibus-engines.table-others</literal>): Various table-based input methods.</para></listitem>
+</itemizedlist>
+</section>
+
+<section><title>Fcitx</title>
+
+<para>Fcitx is an input method framework with extension support. It has three built-in Input Method Engine, Pinyin, QuWei and Table-based input methods.</para>
+<para>The following snippet can be used to configure Fcitx:</para>
+
+<programlisting>
+i18n.inputMethod = {
+  enabled = "fcitx";
```

Comment:
The motivation was that only one method can be enabled at a time and it was feeling cleaner to have one option to chose which input method to enable, than one boolean `enable` for each with checks that others input methods are not enabled.

The `enabled` option is defined as:

```
  options = {
    i18n.inputMethod = {
      enabled = mkOption {
        type    = types.nullOr (types.enum [ "ibus" "fcitx" "nabi" "uim" ]);
        default = null;
        example = "fcitx";
        description = ''
          Select the enabled input method. Input methods is a software to input symbols that are not available on standard input devices.

          Input methods are specially used to input Chinese, Japanese and Korean characters.

          Currently the following input methods are available in NixOS:

          <itemizedlist>
          <listitem><para>ibus: The intelligent input bus, extra input engines can be added using <literal>i18n.inputMethod.ibus.engines</literal>.</para></listitem>
          <listitem><para>fcitx: A customizable lightweight input method, extra input engines can be added using <literal>i18n.inputMethod.fcitx.engines</literal>.</para></listitem>
          <listitem><para>nabi: A Korean input method based on XIM. Nabi doesn't support Qt 5.</para></listitem>
          <listitem><para>uim: The universal input method, is a library with a XIM bridge. uim mainly support Chinese, Japanese and Korean.</para></listitem>
          </itemizedlist>
        '';
      };
    };
  };
```

It also happen to be a useful central point to manage input methods related files like gtk cache files in #14568.